citations/citations-impl/impl/src/java/org/sakaiproject/citation/impl/DbCitationService.java

```diff
@@ -913,6 +915,24 @@ protected void deleteCitation(Citation edit)
         }
 
 		/* (non-Javadoc)
+		* @see org.sakaiproject.citation.impl.BaseCitationService.Storage#deleteCitationCollectionOrder(rg.sakaiproject.citation.api.CitationCollectionOrder))
+		*/
+		protected void deleteCitationCollectionOrder(CitationCollectionOrder citationCollectionOrder)
+		{
+			String statement = "delete from " + m_collectionOrderTableName + " where (" + m_collectionTableId + " = ? AND " +
+					"LOCATION = ? AND SECTION_TYPE = ? AND VALUE = ? )";
```

Comment:
Don't you just want to delete anything at the location, if the value and type don't match does that matter?